ShopLens/ShopLensForms/Controllers/MainController.cs

```diff
@@ -0,0 +1,165 @@
+            _voiceRecognizer.AddCommand(exitCmd, CommandRecognized_Exit);
+            _voiceRecognizer.StartVoiceRecognition();
+        }
+
+        /// <summary> Calls method when someone says a certain command. </summary>
+        /// <remarks>
+        /// This if statement makes sure the method that must be executed when the specified command is recognized
+        /// is called within the GUI thread. For information see https://stackoverflow.com/a/10170699.
+        /// </remarks>
+        private void CommandRecognized_Hello(object sender, EventArgs e)
+        {
+            if (_introForm.InvokeRequired)
+            {
+                _introForm.BeginInvoke(new MethodInvoker(() => _introForm.Enter_btn_Click(sender, e)));
+            }
+            else
+            {
+                _introForm.Enter_btn_Click(sender, e);
+            }
+        }
+
+        /// <inheritdoc cref="CommandRecognized_Hello(object, EventArgs)"/>
+        private void CommandRecognized_WhatIsThis(object sender, EventArgs e)
+        {
+            if (_shopLens.InvokeRequired)
+            {
+                _shopLens.BeginInvoke(new MethodInvoker(() => _shopLens.WhatIsThis_btn_Click(sender, e)));
+            }
+            else
+            {
+                _shopLens.WhatIsThis_btn_Click(sender, e);
+            }
+        }
+
+        /// <inheritdoc cref="CommandRecognized_Hello(object, EventArgs)"/>
+        private void CommandRecognized_Start(object sender, EventArgs e)
+        {
+            if (_shopLens.InvokeRequired)
+            {
+                _shopLens.BeginInvoke(new MethodInvoker(() => _shopLens.Start_btn_Click(sender, e)));
+            }
+            else
+            {
+                _shopLens.Start_btn_Click(sender, e);
+            }
+        }
+
+        /// <inheritdoc cref="CommandRecognized_Hello(object, EventArgs)"/>
+        private void CommandRecognized_Exit(object sender, EventArgs e)
+        {
+            if (_shopLens.InvokeRequired)
+            {
+                _shopLens.BeginInvoke(new MethodInvoker(() => _shopLens.Exit_btn_Click(sender, e)));
+            }
+            else
+            {
+                _shopLens.Exit_btn_Click(sender, e);
+            }
+        }
+
+        /// <summary>
+        /// Uses a text voicer object to voice a message.
+        /// </summary>
+        /// <param name="seeMessage">The string of a message to be voiced.</param>
+        public void TextVoicerVoiceMessage(string seeMessage)
+        {
+            _textVoicer.SayMessage(seeMessage);
+        }
+
+        /// <summary>
+        /// Uses an image classifier object to return a Dictionary of labels and their
+        /// probabilities of being in a particular image.
+        /// </summary>
+        /// <param name="imgArray">A byte array of an image 
+        /// to be classified by the image classifier object.</param>
+        /// <returns></returns>
```

Comment:
Dictionary<string, float> is a bit arbitrary, so it would be great to specify that it returns {label name: float (range 0-1)}

Reply:
That's a good point.